node/silkworm/stagedsync/stage_execution.cpp

```diff
@@ -31,120 +30,169 @@
-        BlockNum block_num{db::stages::read_stage_progress(*txn, db::stages::kExecutionKey) + 1};
-        if (block_num > max_block) {
-            log::Error() << "Stage progress is " << (block_num - 1) << " which is <= than requested block_to "
-                                << max_block;
-            return StageResult::kInvalidRange;
-        }
-
-        // Execution needs senders hence we need to check whether sender's stage is
-        // at least at max_block as set above
-        const BlockNum max_block_senders{db::stages::read_stage_progress(*txn, db::stages::kSendersKey)};
-        if (max_block > max_block_senders) {
-            log::Error() << "Sender's stage progress is " << (max_block_senders)
-                                << " which is <= than requested block_to " << max_block;
-            return StageResult::kMissingSenders;
-        }
-
-        StopWatch sw{};
-        (void)sw.start();
-
-        for (; block_num <= max_block; ++block_num) {
-            res = execute_batch_of_blocks(*txn, chain_config.value(), max_block, batch_size, block_num, prune_from);
-            if (res != StageResult::kSuccess) {
-                return res;
-            }
-
-            db::stages::write_stage_progress(*txn, db::stages::kExecutionKey, block_num);
-
-            txn.commit();
+StageResult Execution::unwind(db::RWTxn& txn, BlockNum to) {
+    (void)txn;
+    (void)to;
+    throw std::runtime_error("Not yet implemented");
+}
 
-            (void)sw.lap();
-            log::Info() << (block_num == max_block ? "All blocks" : "Blocks") << " <= " << block_num
-                               << " committed"
-                               << " in " << StopWatch::format(sw.laps().back().second);
-        }
-    } catch (const mdbx::exception& ex) {
-        log::Error() << "DB Error " << ex.what() << " in stage_execution";
-        return StageResult::kDbError;
-    } catch (const std::exception& ex) {
-        log::Error() << "Unexpected error " << ex.what() << " in stage execution";
-        return StageResult::kUnexpectedError;
+StageResult Execution::prune(db::RWTxn& txn) {
+    (void)txn;
+    throw std::runtime_error("Not yet implemented");
+};
+
+std::vector<std::string> Execution::get_log_progress() {
+    std::unique_lock progress_lock(progress_mtx_);
+    auto now{std::chrono::steady_clock::now()};
+    auto elapsed{now - lap_time_};
+    lap_time_ = now;
+    auto elapsed_seconds = static_cast<uint32_t>(std::chrono::duration_cast<std::chrono::seconds>(elapsed).count());
+    if (!elapsed_seconds || !processed_blocks_) {
+        return {"block", std::to_string(block_num_)};
     }
-
-    return res;
+    auto speed_blocks = processed_blocks_ / elapsed_seconds;
+    auto speed_transactions = processed_transactions_ / elapsed_seconds;
+    auto speed_mgas = processed_mgas_ / elapsed_seconds;
+    processed_blocks_ = 0;
+    processed_transactions_ = 0;
+    processed_mgas_ = 0;
+    progress_lock.unlock();
```

Comment:
`progress_lock.unlock()` is unnecessary

Reply:
Well ... actually it is : it releases the lock **before** the function continues in building a vector with 4 to string conversions.
Yes is a matter of nanoseconds but is clear that the developer releases the lock ASAP

Reply:
fair enough